dbt_server/state_test.py

```diff
@@ -0,0 +1,464 @@
+from dbt_server.exceptions import StateNotFoundException
+from dbt_server.state import LAST_PARSED
+from dbt_server.state import CachedManifest
+from dbt_server.state import StateController
+from dbt_server.views import ParseArgs
+from unittest import TestCase
+from unittest import mock
+
+TEST_STATE_ID = "test_state"
+TEST_PROJECT_PATH = "test_project_path"
+TEST_ROOT_PATH = "test_root_path"
+TEST_MANIFEST_FILE = "test_root_path/manifest.msgpack"
+TEST_PARTIAL_PARSE_FILE = "partial.msgpack"
+TEST_MANIFEST = "test_manifest"
+TEST_MANIFEST_SIZE = 10
+TEST_CONFIG = "test_config"
+TEST_SQL_PARSER = "test_sql_parse"
+TEST_ARGS = 1
+TEST_VALUE = 10
+TEST_QUERY = "test_query"
+TEST_TASK_ID = "test_task_id"
+TEST_DB = "db"
+TEST_CALLBACK_URL = "test_callback_url"
+TEST_COMMAND = ["testa", "testb"]
+
+
+@mock.patch(
+    "dbt_server.services.dbt_service.get_sql_parser", return_value=TEST_SQL_PARSER
+)
+def _set_test_cache(state_id, _):
+    """Helper sets global singleton manifest cache."""
+    LAST_PARSED.set_last_parsed_manifest(
+        state_id,
+        TEST_PROJECT_PATH,
+        TEST_ROOT_PATH,
+        TEST_MANIFEST,
+        TEST_MANIFEST_SIZE,
+        TEST_CONFIG,
+    )
+
+
+def _get_test_cache(state_id: str, project_path: str = TEST_PROJECT_PATH):
+    """Helper returns a CachedManifest object with given `state_id` and
+    `project_path`.
+    """
+    return CachedManifest(
+        state_id,
+        project_path,
+        TEST_ROOT_PATH,
+        TEST_MANIFEST,
+        TEST_MANIFEST_SIZE,
+        TEST_CONFIG,
+        TEST_SQL_PARSER,
+    )
+
+
+class TestCachedManifest(TestCase):
+    def setUp(self) -> None:
+        LAST_PARSED.reset()
```

Comment:
I think it might make sense to instead call `_set_test_cache()` in `setUp`, and then call this reset in `tearDown()` -- thoughts?

Reply:
Agree we should reset in tearDown, some test cases are against empty cache so I don't put _set_test_cache in setup, instead just call it in each test case.